### PR TITLE
Cleaned up the layout of the Property Editor and added some more functionality to it

### PR DESCRIPTION
**Pull request by roeyb1**
Also renamed the SceneGraph to SceneHierarchy everywhere because the naming was inconsistent

### CODE REVIEW

File: ScarletEditor/Sources/PropertyEditor/Private/PropertyEditor.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #include "Editor.h"
 #include "ECS.h"
 #include "Widgets.h"
+#include "Components/RigidBodyComponent.h"
```
Comment (RobinLmn) [inline ScarletEditor/Sources/PropertyEditor/Private/PropertyEditor.cpp:6]: shouldn't you include Achilles.h ?

File: ScarletEditor/Sources/SceneHierarchy/Private/SceneHierarchy.cpp [hunk 2]
```diff
@@ -30,12 +28,12 @@ namespace ScarletEngine
...
 		ZoneScoped
-		ImGuiTreeNodeFlags BaseFlags = ImGuiTreeNodeFlags_OpenOnArrow | ImGuiTreeNodeFlags_OpenOnDoubleClick | ImGuiTreeNodeFlags_SpanAvailWidth;
+		const ImGuiTreeNodeFlags BaseFlags = ImGuiTreeNodeFlags_OpenOnArrow | ImGuiTreeNodeFlags_OpenOnDoubleClick | ImGuiTreeNodeFlags_SpanAvailWidth;
 		for (const auto& [ID, EntItem] : Items)
 		{
-			ImGui::PushID((uint32_t)ID);
+			ImGui::PushID(static_cast<uint32_t>(ID));
 			char Buffer[128];
-			snprintf(Buffer, 128, "%s %s", ICON_MD_CUBE, EntItem->GetDisplayString());
+			snprintf(Buffer, 128, "%s   %s", ICON_MD_CUBE, EntItem->GetDisplayString());
```
Comment (RobinLmn) [inline ScarletEditor/Sources/SceneHierarchy/Private/SceneHierarchy.cpp:36]: why extra space?

**Review by RobinLmn — APPROVED** [review]

File: ScarletEditor/Sources/PropertyEditor/Private/PropertyEditor.cpp (hunk 1 above)
Comment (roeyb1) [reply ScarletEditor/Sources/PropertyEditor/Private/PropertyEditor.cpp:6]: I don't need all of Achilles, just the rigidbody component. In principal you should be as specific as possible when including.

File: ScarletEditor/Sources/SceneHierarchy/Private/SceneHierarchy.cpp (hunk 2 above)
Comment (roeyb1) [reply ScarletEditor/Sources/SceneHierarchy/Private/SceneHierarchy.cpp:36]: Makes the scene hierarchy a little less compressed, adds some space between the icon and the label.

**Pull request merged**